### PR TITLE
test(contracts): re-organize smart contract tests in separate files

### PR DESCRIPTION
**Pull request by ctrlc03**
Currently all tests related to a Poll were inside one file (MACI.test) - these tests have been moved into their respective files, re organized and cleaned up, and a small number of test cases were added which were missing.

re #843

### CODE REVIEW

**Review by baumstern — CHANGES_REQUESTED** [review]
Great work! Just left a nitpick

File: contracts/contracts/Poll.sol [hunk 1]
```diff
@@ -98,7 +98,7 @@ contract PollFactory is Params, IPubKey, Ownable, PollDeploymentParams {
 contract Poll is Params, Utilities, SnarkCommon, Ownable, PollDeploymentParams, EmptyBallotRoots {
   using SafeERC20 for ERC20;
 
-  bool internal isInit = false;
+  bool public isInit;
```
[the remark this answers is not in the document]
Comment (baumstern) [reply contracts/contracts/Poll.sol:101]: Isn't L84:85 is redundant?:
https://github.com/privacy-scaling-explorations/maci/blob/050e7887fdf381b6d715abec87eab8d0567b1d03/contracts/tests/Poll.test.ts#L84-L85

L87 checks same condition what L84:85 does:
https://github.com/privacy-scaling-explorations/maci/blob/050e7887fdf381b6d715abec87eab8d0567b1d03/contracts/tests/Poll.test.ts#L87

File: contracts/tests/Poll.test.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    }
+    const message = new Message(BigInt(1), messageData);
+    const padKey = new PubKey([
+      BigInt("10457101036533406547632367118273992217979173478358440826365724437999023779287"),
+      BigInt("19824078218392094440610104313265183977899662750282163392862422243483260492317"),
+    ]);
+    maciState.polls[pollId].publishMessage(message, padKey);
+  });
+
+  it("should not be possible to init the Poll contrac twice", async () => {
```
Comment (kittybest) [inline contracts/tests/Poll.test.ts:83]: typo: contract

File: contracts/contracts/Poll.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (ctrlc03) [reply contracts/contracts/Poll.sol:101]: sure I see what you mean, fixing now

**Review by baumstern — APPROVED** [review]
Looks good to me! I'll leave it open for @kittybest to take a peek

**Review by kittybest — APPROVED** [review]

**Pull request merged**